### PR TITLE
REL: 4.2.0

### PR DESCRIPTION
**Pull request by stefan6419846**
Please do not merge after approval to set the correct commit message.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,64 @@
...
+
+### Testing (TST)
+- Allow loading truncated images if required (#2586)
+- Fix download issues from #2562 (#2578)
+- Improve test_get_contents_from_nullobject to show real use-case (#2524)
+- Add missing test annotations (#2507)
+
+### Other
+- FIX: Extract text in layout mode without finding resources (#2555)
+- FIX: Robustify stream extraction (#2526)
```
Comment (MartinThoma) [inline CHANGELOG.md:58]: Those two should be in the "Bug Fixes (BUG)"  section

Comment (stefan6419846) [reply CHANGELOG.md:58]: We should probably have a look at the `FIX` prefix for further releases as well to not have them in "Other" each time.

**Comment by MartinThoma** [comment]
I just noticed a disadvantage of the new workflow:

In the old workflow I removed parts from the changelog that were likely not interesting to people who just wondered if they should update pypdf or not. Especially DOC (documentation), TST (tests), and internal changes (e.g. CI pipeline, pre-commit, ...) were things that I removed.

**Review by MartinThoma — DISMISSED** [review]

**Review by pubpub-zz — APPROVED** [review]

**Comment by pubpub-zz** [comment]
@stefan6419846 I will let you finish

**Comment by stefan6419846** [comment]
> In the old workflow I removed parts from the changelog that were likely not interesting to people who just wondered if they should update pypdf or not. Especially DOC (documentation), TST (tests), and internal changes (e.g. CI pipeline, pre-commit, ...) were things that I removed.

This is something we could still do with the new workflow. I just did not yet bother enough about cleaning/prioritizing them ;)

File: CHANGELOG.md (hunk 1 above)
Comment (stefan6419846) [reply CHANGELOG.md]: I have moved them accordingly (2526 was indeed a ROB change and I just missed to fix the category during the merge).

**Review by pubpub-zz — APPROVED** [review]

**Comment by stefan6419846** [comment]
By the way: `FIX` has never been a valid prefix. @pubpub-zz just uses them as commit message and merging such single-commit PRs into the `main` branch causes this oddity if we do not change the default merge message, although the PR title is correct.

**Comment by stefan6419846** [comment]
Damn, the automatic release has some issues again: https://github.com/py-pdf/pypdf/actions/runs/8588793466/job/23534111592

**Comment by MartinThoma** [comment]
I'll manually release 4.2

**Comment by MartinThoma** [comment]
Ok, pypdf==4.2.0 is on pypi :-)

**Comment by MartinThoma** [comment]
@stefan6419846 What do you think about splitting the current release workflow into:

1. REL commit detected -> create git tag
2. New git tag -> create github release page
3. New git tag -> build+upload to PyPI

This way we could be more robust if issues occur in future.

**Comment by stefan6419846** [comment]
@MartinThoma Thanks, I have just created the GitHub release for the tag manually as well.

Regarding splitting: I am not sure whether this is too much, but at least decoupling GitHub and PyPI might be a good idea (some days ago, PyPI blocked new uploads due to a huge amount of spam for some period).

I would instead propose to make the actual value retrieval more stable by not relying on Bash functionality any more, but using a Python script for the version and message as people tend to be more fluent with Python and which can actually be tested automatically by integration tests.

**Pull request merged**